deltachat-time/src/lib.rs

```diff
@@ -1,9 +1,11 @@
 #![allow(missing_docs)]
 
-use std::sync::RwLock;
 use std::time::{Duration, SystemTime};
 
-static SYSTEM_TIME_SHIFT: RwLock<Duration> = RwLock::new(Duration::new(0, 0));
+use tokio::sync::{Mutex, MutexGuard, RwLock};
+
+static SYSTEM_TIME_SHIFT_MTX: Mutex<()> = Mutex::const_new(());
```

Comment:
> The priority policy of Tokio’s read-write lock is fair (or [write-preferring](https://en.wikipedia.org/wiki/Readers%E2%80%93writer_lock#Priority_policies)), in order to ensure that readers cannot starve writers.

That's why this mutex is needed, otherwise `now()` would wait for `shift()` to be done thus deadlocking.